// @shelby/client-sdk/dist/chunk-M3CP5TFP.mjs
// src/core/rpc-responses.ts
import { z } from "zod";
var ChallengeResponseSchema = z.object({
  challenge: z.string(),
  expiresAt: z.number()
});
var MultipartUploadStatusResponseSchema = z.object({
  uploadId: z.string(),
  completedParts: z.array(z.number()),
  partSize: z.number(),
  nParts: z.number(),
  uploadedBytes: z.number()
});
var StartMultipartUploadResponseSchema = z.object({
  uploadId: z.string()
});
var UploadPartResponseSchema = z.object({
  success: z.literal(true)
});
var CompleteMultipartUploadResponseSchema = z.object({
  success: z.literal(true)
});
var RPCErrorResponseSchema = z.object({
  error: z.string()
});
var StaleMicropaymentErrorResponseSchema = z.object({
  error: z.string().optional(),
  storedMicropayment: z.string().optional()
});

export {
  ChallengeResponseSchema,
  MultipartUploadStatusResponseSchema,
  StartMultipartUploadResponseSchema,
  UploadPartResponseSchema,
  CompleteMultipartUploadResponseSchema,
  RPCErrorResponseSchema,
  StaleMicropaymentErrorResponseSchema
};
